cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/LocalizationUtils.java

```diff
@@ -42,6 +48,43 @@
 public final class LocalizationUtils {
   private static final Logger LOG = LoggerFactory.getLogger(LocalizationUtils.class);
 
+  public static List<File> getExtraJars(CConfiguration cConf) {
+    String extraJarConf = cConf.get(Constants.AppFabric.EXTRA_JARS);
+    List<String> extraJars = extraJarConf == null ?
+      new ArrayList<String>() : Arrays.asList(extraJarConf.split(","));
+    List<File> jarFiles = new ArrayList<>();
+    for (String jarPath : extraJars) {
+      jarFiles.add(new File(jarPath));
+    }
+    return jarFiles;
+  }
+
+  public static  List<String> getExtraJarNames(CConfiguration cConf) {
+    List<String> jars = new ArrayList<>();
+    for (File jar : getExtraJars(cConf)) {
+      jars.add(jar.getName());
+    }
+    return jars;
+  }
+
+  /**
+   * Add the extra jars path set in the {@link CConfiguration} in a map.
+   */
+  public static void addExtraJars(Map<String, LocalizeResource> resourcesToLocalize, CConfiguration cConf) {
+    for (File jar : getExtraJars(cConf)) {
+      resourcesToLocalize.put(jar.getName(), new LocalizeResource(jar));
+    }
+  }
+
+  /**
+   * Add the extra jars set in the {@link CConfiguration} in a list .
+   */
+  public static void addLocalizedExtraJars(List<LocalizeResource> resourcesToLocalize, CConfiguration cConf) {
```

Comment:
And BTW, as @sagarkapare suggested, this method is very similar to the other one. Don't introduce method like this because of a specific need from the caller side.